Tests/NetworkServiceTests/MockURLProtocol.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+
+/// A custom URL protocol used for mocking network responses during testing.
+class MockURLProtocol: URLProtocol {
+    /// Handles incoming requests and returns mocked responses.
+    static var requestHandler: ((URLRequest) throws -> (HTTPURLResponse, Data?))?
+
+    override class func canInit(with _: URLRequest) -> Bool {
+        return true
+    }
+
+    override class func canonicalRequest(for request: URLRequest) -> URLRequest {
+        return request
+    }
+
+    override func startLoading() {
+        guard let handler = MockURLProtocol.requestHandler else {
+            fatalError("MockURLProtocol.requestHandler is not set.")
+        }
+
+        do {
+            let (response, data) = try handler(request)
+
+            if let data = data {
+                client?.urlProtocol(self, didReceive: response, cacheStoragePolicy: .notAllowed)
+                client?.urlProtocol(self, didLoad: data)
+            }
+
+            client?.urlProtocolDidFinishLoading(self)
```

Comment:
What if the data is `nil`? Should we send a `URLResponse` in these cases ?
Something like:
```suggestion
                        if let data {
                client?.urlProtocol(self, didLoad: data)
            }

            client?.urlProtocol(self, didReceive: response, cacheStoragePolicy: .notAllowed)
            client?.urlProtocolDidFinishLoading(self)
```

Reply:
Ok, fixed with 203e327